en/setup/vscode.md

```diff
@@ -0,0 +1,55 @@
+# Visual Studio Code IDE
+
+Visual Studio Code (VSCode) is a powerful cross-platform source code editor/IDE that can be used for PX4 development on Linux and macOS (Windows support coming soon).
+
+PX4 has been set up with all the definition files that you need to install needed VSCode extensions, and and start developing.
+This topic explains how to get started.
+
+<!-- 
```

Comment:
Some questions here. Normally it is good to explain why we're suggesting this IDE, in particular if there is a change - ie from Qt Creator
1. Are we _recommending_ this over Qt Creator, and if so why?

Reply:
It mostly comes down to richer extension ecosystem and simple in tree configuration.

We have good support for the main things like C/C++ (with solid cmake integration), but also python, jinja2, ROS messages, and even UAVCAN dsdl. Beyond that the important difference is that the configuration is simple and lives in the PX4/Firmware tree (https://github.com/PX4/Firmware/tree/master/.vscode). That's the somewhat mundane, yet critical difference that makes this a good solution for most people. It's possible to configure other editors and IDEs to do most of this, but this one is ready to go for a new user in minutes, not an afternoon of manual setup.

![image](https://user-images.githubusercontent.com/84712/57471285-d784bd80-7258-11e9-88fd-387b734532bb.png)